File: Solutions/1464/1464.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/8/26 08:23
# File: 1464.py
# Desc: 


class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        # a=0
        # b=0
        # for i in nums:
        #     if i >= a:
        #         if i >= b:
        #             a = b
        #             b = i
        #         else:
        #             a = i
        # return (a-1)*(b-1)
        nums.sort()
        return (nums[-1]-1) * (nums[-2]-1)

File: Solutions/1464/1464.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1464.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-08-26 08:25:27
***************************************************************************************************/


int maxProduct(int* nums, int numsSize){
    int a=0, b=0;
    for (int i=0; i<numsSize; i++) {
        if (nums[i] >= a) {
            if (nums[i] >= b) {
                a = b;
                b = nums[i];
            }
            else a = nums[i];
        }
    }
    return (a-1) * (b-1);
}
